neuraxle/metaopt/callbacks.py

```diff
@@ -0,0 +1,250 @@
+                return True
+        return False
+
+class IfLastStep(MetaCallback):
+    def call(self, trial: Trial, epoch_number: int, total_epochs: int, input_train: DataContainer,
+             pred_train: DataContainer, input_val: DataContainer, pred_val: DataContainer, is_finished_and_fitted: bool):
+        if epoch_number == total_epochs - 1:
+            self.wrapped_callback.call(
+                trial,
+                epoch_number,
+                total_epochs,
+                input_train,
+                pred_train,
+                input_val,
+                pred_val,
+                is_finished_and_fitted
+            )
+            return True
+        return False
+
+
+class StepSaverCallback(MetaCallback):
+    def call(self, trial: Trial, epoch_number: int, total_epochs: int, input_train: DataContainer,
+             pred_train: DataContainer, input_val: DataContainer, pred_val: DataContainer, is_finished_and_fitted: bool):
+        trial.save_model()
+        return False
+
+
+class CallbackList(BaseCallback):
+    def __init__(self, callbacks, print_func: Callable = None):
+        self.callbacks = callbacks
+        if print_func is None:
+            print_func = print
+        self.print_func = print_func
+
+    def call(self, trial: Trial, epoch_number: int, total_epochs: int, input_train: DataContainer,
+             pred_train: DataContainer, input_val: DataContainer, pred_val: DataContainer, is_finished_and_fitted: bool):
+        is_finished_and_fitted = False
+        for callback in self.callbacks:
+            try:
+                if callback.call(
+                        trial=trial,
+                        epoch_number=epoch_number,
+                        total_epochs=total_epochs,
+                        input_train=input_train,
+                        pred_train=pred_train,
+                        input_val=input_val,
+                        pred_val=pred_val,
+                        is_finished_and_fitted=is_finished_and_fitted
+                ):
+                    is_finished_and_fitted = True
+            except Exception as error:
+                track = traceback.format_exc()
+                self.print_func(track)
+        return is_finished_and_fitted
+
+
+class MetricCallback(BaseCallback):
+    def __init__(self, name: str, metric_function: Callable, higher_score_is_better: bool):
+        self.name = name
+        self.metric_function = metric_function
+        self.higher_score_is_better = higher_score_is_better
+
+    def call(self, trial: Trial, epoch_number: int, total_epochs: int, input_train: DataContainer,
+             pred_train: DataContainer, input_val: DataContainer, pred_val: DataContainer, is_finished_and_fitted: bool):
+        train_score = self.metric_function(pred_train.data_inputs, pred_train.expected_outputs)
+        validation_score = self.metric_function(pred_val.data_inputs, pred_val.expected_outputs)
+
+        trial.set_metric_results_train(
```

Comment:
I'd rename set to add or update as we don't overwrite and rather register a new score.